mate-screenshot/src/mate-screenshot.c

```diff
@@ -1180,10 +1180,15 @@ load_options (void)
   /* Find various dirs */
   last_save_dir = g_settings_get_string (settings,
                                          LAST_SAVE_DIRECTORY_KEY);
-  if (!last_save_dir || !last_save_dir[0])
+  if (last_save_dir == NULL)
     {
       last_save_dir = get_desktop_dir ();
     }
+  else if (*last_save_dir == '\0')
+    {
+      g_free (last_save_dir);
+      last_save_dir = get_desktop_dir ();
+    }
```

Comment:
What about:
```C
if (last_save_dir && !last_save_dir[0])
  {
    g_free (last_save_dir);
    last_save_dir = NULL;
  }

if (!last_save_dir)
  {
    /* ... */
```
e.g. avoid duplicating `get_desktop_dir()` call so it makes it easier to maintain, and simply consider an empty string the same as `NULL`.

Reply:
Do you want to group `(last_save_dir == NULL)` and `(*last_save_dir == '\0')` ?
```
if ((last_save_dir == NULL) || (*last_save_dir == '\0'))
  {
      g_free (last_save_dir);
      last_save_dir = get_desktop_dir ();
  }
else if (last_save_dir[0] == '~')
```
There is something weird in your proposal.

Reply:
Ok, now I understand you. You want to free the string when it's not null and only has the '\0'

Reply:
In fact `last_save_dir` is never NULL, so we can remove it.